libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -45,6 +47,109 @@ static int lua_micros(lua_State *L) {
     return 1;
 }
 
+static int lua_mavlink_receive(lua_State *L) {
+    check_arguments(L, 0, "receive");
+
+    struct AP_Scripting::mavlink_msg msg;
+    ObjectBuffer<struct AP_Scripting::mavlink_msg> *input = AP::scripting()->mavlink_data.input;
+
+    if (input == nullptr) {
+        luaL_error(L, "Never subscribed to a message");
+    }
+
+    if (input->pop(msg)) {
+        luaL_Buffer b;
+        luaL_buffinit(L, &b);
+        luaL_addlstring(&b, (char *)&msg.msg, sizeof(msg.msg));
+        luaL_pushresult(&b);
+        lua_pushinteger(L, msg.chan);
+        return 2;
+    } else {
+        // no MAVLink to handle, just return no results
+        return 0;
+    }
+}
+
+static int lua_mavlink_register_msgid(lua_State *L) {
+    check_arguments(L, 1, "register_msgid");
+    luaL_checkstack(L, 1, "Out of stack");
+    
+    const int msgid = luaL_checkinteger(L, -1);
+    luaL_argcheck(L, ((msgid >= 0) && (msgid < (1 << 24))), 1, "msgid out of range");
+
+    struct AP_Scripting::mavlink &data = AP::scripting()->mavlink_data;
+
+    if (data.input == nullptr) {
+        { // WITH_SEMAPHORE cannot be used in the same scope as a luaL_error (or any thing else that jumps away)
+            WITH_SEMAPHORE(data.sem);
+            data.input = new ObjectBuffer<struct AP_Scripting::mavlink_msg>(AP_Scripting::mavlink_input_queue_size);
+        }
+        if (data.input == nullptr) {
+            return luaL_error(L, "Unable to allocate MAVLink buffer");
+        }
+    }
+
+    // check that we aren't currently watching this ID
+    for (int id : data.accept_msg_ids) {
+        if (id == msgid) {
+            lua_pushboolean(L, false);
```

Comment:
Don't we want to return true if already watching this ID? If we did that the function might as well not return anything tho.